db/models.py

```diff
@@ -46,3 +53,76 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class User(AbstractUser):
+    pass
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(to=settings.AUTH_USER_MODEL,
+                             on_delete=models.CASCADE,
+                             related_name="orders")
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(to=MovieSession,
+                                      on_delete=models.CASCADE,
+                                      related_name="tickets")
+    order = models.ForeignKey(to=Order,
+                              on_delete=models.CASCADE,
+                              related_name="tickets")
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(fields=["row", "seat", "movie_session"],
+                             name="row_seat_session_unique")
+        ]
+
+    def __str__(self) -> str:
+        return (
+            f"{self.movie_session.movie.title} "
+            f"{self.movie_session.show_time} "
+            f"(row: {self.row}, seat: {self.seat})"
+        )
+
+    def clean(self) -> None:
+        cinema_hall = self.movie_session.cinema_hall
+        if self.row > cinema_hall.rows:
+            raise ValidationError(
+                {
+                    "row": [f"row number must be in available range: "
+                            f"(1, rows): "
+                            f"(1, {cinema_hall.rows})"]
+                }
+            )
+        if self.seat > cinema_hall.seats_in_row:
+            raise ValidationError(
+                {
+                    "seat": [
+                        f"seat number must be in available range: "
+                        f"(1, seats_in_row): "
+                        f"(1, {cinema_hall.seats_in_row})"
+                    ]
+                }
+            )
+
+    def save(self,
+             force_insert: Any = False,
+             force_update: Any = False,
+             using: Any = None,
+             update_fields: Any = None) -> None:
+        self.full_clean()
+        return super(Ticket, self).save(force_insert,
```

Comment:
In recent python updates there is no need to write `(Ticket, self)` in `super()`